### ISSUE TITLE
Having '.md' anywhere inside new note name will break title generation

### ISSUE THREAD

**Issue by LoPawlie** [dendronhq/dendron#856]
## Please select the area your Bug applys to. (Multiple selections are Possible)

- [ ] Onboard. Initial vault setup and import/export pods 
- [x] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [ ] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Describe the bug

When creating a new note, any hierarchy element that starts with 'md' (and is not a domain) will break the auto title generation by using the previous hierarchy element to title the note/stub that starts with 'md' and any subsequent children instead of using the proper ones.

## To Reproduce

1. Go to Lookup.
2. Write `learn.mdn.css.text`
3. Frontmatter shows `title: Learn` for the new note.
3. Expand tree view to see the whole new hierarchy titled 'learn'
    ```
    .
    └── learn
        └── learn
            └── learn
                └── learn
    ```

## Expected behavior

The note would be titled 'Text' and the tree view would look like this:
```
.
└── learn
    └── mdn
        └── css
            └── text
```

## Desktop

- OS: macOS 10.15
- VSCode: 1.57.1
- Dendron: 0.47.0

## Dendron Log file

Please attach the output of `>Dev:Dendron: Open Logs` here

[dendron.log](https://github.com/dendronhq/dendron/files/6698655/dendron.log)

**Comment by kevinslin** [comment]
That's a bug. Thanks for catching this. We'll look into this for the upcoming release

**Comment by chmac** [comment]
I think #1972 is probably related to this. I'd guess that there's a regex somewhere which is `\.md` instead of `\.md$`.

@kevinslin Any pointers as to where this issue might be in the codebase? I could have a look and see if I can find the faulty regex if you're able to offer a general pointer, like which package this code would be in, etc.

**Comment by Harshita-mindfire** [comment]
Hi, this has been fixed with Dendron v0.82. Please let us know if it continues to happen, otherwise, this issue will close in a few days.

**Comment by chmac** [comment]
Looks fixed to me.
